src/EnergyPlus/LowTempRadiantSystem.cc

```diff
@@ -2287,16 +2341,90 @@ namespace LowTempRadiantSystem {
 
-            } else {
+                ElecRadSys(RadSysNum).OperatingMode = NotOperating;
+                
             }
         }
 
-        OperatingMode = NotOperating; // System is not operating or can't operate; will be reset elsewhere, if necessary
     }
 
+    void HydronicSystemBaseData::updateOperatingModeHistory()
+    {
+        // Since this is only called when the operating mode is something other than "not operating",
+        // the status from the previous system time step is what it did in the last or previous time step.
+        // So, we can update the last status of the system using this information before reseting things
+        // to "not operating".
+        this->lastOperatingMode = this->OperatingMode;
+        
+        if (DataGlobals::BeginDayFlag) {
+            // The begin day flag is set which mean this is the first time step of the day.
+            // This also means that the previous time step was the last time step of yesterday.
+            // So, the day should be the previous day, the hour should bethe last hour of the
+            // day, and the time step should be the last time step.
+            this->lastDayOfSim  = DataGlobals::DayOfSim - 1;
+            this->lastHourOfDay = int(DataGlobals::HoursInDay);
+            this->lastTimeStep  = DataGlobals::NumOfTimeStepInHour;
+        } else if (DataGlobals::BeginHourFlag) {
+            // It's not the beginning of the day but it is the beginning of an hour other than
+            // the first hour.  This means that the previous time step was the previous hour of
+            // today in the last time step.  So, the day should be the current day, the hour should
+            // be the previous hour, and the time step should be the last time step.
+            this->lastDayOfSim  = DataGlobals::DayOfSim;
+            this->lastHourOfDay = DataGlobals::HourOfDay - 1;
+            this->lastTimeStep  = DataGlobals::NumOfTimeStepInHour;
+        } else if (DataGlobals::BeginTimeStepFlag) {
+            // It's neither the beginning of the day nor the beginning of an hour but it is the start
+            // of a time step other than the first time step in the hour.  So, the day should be the
+            // current day, the hour should be the current hour, and the time step should be the
+            // previous time step.
+            this->lastDayOfSim  = DataGlobals::DayOfSim;
+            this->lastHourOfDay = DataGlobals::HourOfDay;
+            this->lastTimeStep  = DataGlobals::TimeStep-1;
+        } else {
+            // It's not the beginning of the day, hour, or time step so the "last" value is simply the
+            // same as the current value.  Note that these parameters only track down to the zone time
+            // step level and will make decisions based on that.
+        this->lastDayOfSim  = DataGlobals::DayOfSim;
+        this->lastHourOfDay = DataGlobals::HourOfDay;
+        this->lastTimeStep  = DataGlobals::TimeStep;
```

Comment:
This should be indented.

Reply:
You're right.  I totally missed that one.  Usually I'm pretty consistent with that, but clearly I didn't indent this properly here.